### DISCUSSION TITLE
Yarn 3 is giving me brain worms

### DISCUSSION THREAD

**Discussion by Poolshark** [yarnpkg/berry#4267] [Q&A] [answered]
Hey! I've been using Yarn for ages now and about 3 months ago I tried to migrate existing, working projects to Yarn 3. I've read articles, migration guides - no luck! It just doesn't work!

Then I thought, maybe if I set up a fresh project with Yarn 3 and copy the source files one by one it might work - no luck either. I really don't know why and to be honest I don't have time to figure out what is not working here but I guess I keep my hands off Yarn 2/3/4 or whatever version we have now already...

Couldn't even get a basic react application with relay working - guess the relay compiler does not support PnP 🤷‍♂️

However, if anyone works with Yarn 3, Workspaces, React, Relay, Typescript, GraphQL, GraphQL Codegen - so basically a modern Monorepo setup - and got it working, then please let me know! 😜

Other than that, sorry guys this has really put me off Yarn...

**Comment by devoto13** [comment, marked as answer]
@Poolshark A bit hard to give you advice without any details. But have you tried using [`node-modules` linker](https://yarnpkg.com/configuration/yarnrc#nodeLinker) instead of `pnp` linker to get the same behavior (`node_modules` folders) as with Yarn 1?

**Reply by Poolshark** [reply]
Wow! Thanks for the quick answer! 🙌

I know that giving advice here is impossible. I would not know where to start, my projects are huge and I don't really know which package exactly causes the problems. I narrowed it don to Relay I guess but it could be any of the dependencies which come with it. I even experienced the appearance of a `node_modules` folder after adding some packages relating to relay - maybe it was `babel-plugin-relay` - which made me really curious. Wha's the point of using Yarn 3, including the all mighty `.yarn/cache` folder, but then having all the dependencies again in the `node_modules`? Could it be that maybe most of the package developers simply don't care about PnP?

**Reply by devoto13** [reply]
Yarn 3 being actively maintained compared to Yarn 1 is one reason I guess. There are some fixes and new features (like `yarn dedupe` command).

> Could it be that maybe most of the package developers simply don't care about PnP?

I'm not a Yarn developer. My perception is that Yarn PnP is definitely not for everyone and it does not work well with some packages. I've used it in several small projects for zero-installs, but in all big projects I happily use `node-modules` linker the same as Yarn 1 with `.yarn/cache` in `.gitignore`.

**Reply by devoto13** [reply]
Basically [these](https://yarnpkg.com/getting-started/migration#step-by-step) are the relatively simple migration steps which also point out that you don't have to use PnP if you don't want to.

**Reply by Poolshark** [reply]
Thanks! I've done all that but I will try to go through it again!
